pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -56,6 +313,93 @@ def __init__(self, structure):
             structure (:class:`pyiron.atomistics.structure.atoms.Atoms`): reference Atom structure.
         """
         self._structure = structure
+        self._interstitials = None
+
+    @property
+    def interstitials(self):
+        if self._interstitials is None:
+            raise AssertionError('Initialize interstitials via get_interstitials()')
+        return self._interstitials
```

Comment:
In line with my cosmetic comment above
```suggestion
        if self._interstitials is None:
            self._interstitials = Interstitials(…)
        return self._interstitials
```

I wouldn't really mind dropping this and forcing people to make an instance first and then use, like we do with the neighborhood.

Reply:
Here in this case we have to give a number for `num_neighbors`. I'm not really sure if we should assume a number...

Reply:
One solution would be to make `num_neighbors` an initialization argument for the class. On this line you could also make `variance_buffer` etc class variables and let `positions` also be a property.

Looking back over the `positions` setter/getter alongside `get_positions`, I think I would be more comfortable if `positions` was a read-only property that did the full jazz found in `get_positions` when called -- i.e. make the current `get_positions` arguments class attributes, make that method private, cache it for faster return, and have `positions` call `_get_positions` under the hood passing in the class attributes (so the cache works?).

Take that with a serious grain of salt though, since I don't currently understand why you would want to give people write access to `positions` -- but you really explicitly *do* do that, so I suspect I'm still missing something pretty fundamental here.

Reply:
Ok, I had a couple false starts to commenting on this issue, but I think I understand now what's happening. I agree with Marvin.

Things I don't like right now:
- The discrepancy between `Atoms.get_neighbors` and `Analyze.get_interstitials`
- `Analyze.get_interstitials` returns two different class types (depending on a boolean flag) 
  - and by default it's *not* the type `Interstitials`(?!?!?!)
  - And then after it's been called `Analyze.get_interstitials` and `Analyze.interstitials` return different things
- I feel like this compounds on top of the already confusing interplay of `Interstitials.positions` vs `Interstitials.get_positions`

Reply:
I don't really understand what you guys mean with the discrepancy between `Atoms.get_neighbors` and `Analyse.get_interstitials`, because `get_neighbors` gives the information on the neighbourhood for each atom (i.e. `n_atoms x n_neighbors`), while `get_interstitials` is about giving interstitial positions. In total, I would rather expect `Interstitials` to be like `Atoms` but not like `Neighbors`.

Reply:
Oh you mean like `get_neighbors` returns `Neighbors` and `get_interstitials` doesn't? If that's the case I agree that it's maybe something that has to be changed.

Reply:
`Atoms.get_neighbors` returns type `Neighbors`^. `Analyze.get_interstitials` returns type `np.ndarray`^^. This type discrepancy is my primary objection. I understand this to be Marvin's objection as well; that use should be 

```python
inter = some_struct.analyze.get_interstitials(**kwargs)
my_pos = inter.get_positions()
stein4 = inter.get_steinhardt_parameters(4)
```

etc. And not:

```python
my_pos  = some_struct.analyze.get_insterstitials(**kwargs)
stein4 = some_struct.analyze.interstitials.get_steinhardt_parameters(4)
```

> In total, I would rather expect Interstitials to be like Atoms but not like Neighbors.

I am OK with this sentiment, and I suspect Marvin is too. The trouble I see is that right now, with access via `Analyse` (sorry, just realized I've been applying wrong spelling), `Interstitials` behaves like `Atoms.positions` not `Atoms`...

^ By default; there is also a flag to return some `Tree` object?
^^ Also by default, we do here have a flag to force the return of 
I'm not really thrilled about returning totally different types based on an argument flag, but (a) I recognize there is room on that front for disagreement, and (b) I think it's tangential to this PR and we don't need to resolve it.

Reply:
> Oh you mean like get_neighbors returns Neighbors and get_interstitials doesn't return Interstitials? If that's the case I agree that it's maybe something that has to be changed.

Sorry, missed this second comment during my initial reply. Yep! Cool, I think we're on a sufficiently similar page then.